src/codeflare_sdk/ray/rayjobs/config.py

```diff
@@ -515,3 +515,88 @@ def _build_gcs_ft_options(self) -> Dict[str, Any]:
             }
 
         return gcs_ft_options
+
+    def add_script_volumes(
+        self, configmap_name: str, mount_path: str = "/home/ray/scripts"
+    ):
+        """
+        Add script volume and mount references to cluster configuration.
+
+        Args:
+            configmap_name: Name of the ConfigMap containing scripts
+            mount_path: Where to mount scripts in containers (default: /home/ray/scripts)
+        """
+        # Check if script volume already exists
+        volume_name = "ray-job-scripts"
+        existing_volume = next(
+            (v for v in self.volumes if getattr(v, "name", None) == volume_name), None
+        )
+        if existing_volume:
+            logger.debug(f"Script volume '{volume_name}' already exists, skipping...")
+            return
+
+        # Check if script mount already exists
+        existing_mount = next(
+            (m for m in self.volume_mounts if getattr(m, "name", None) == volume_name),
+            None,
+        )
+        if existing_mount:
+            logger.debug(
+                f"Script volume mount '{volume_name}' already exists, skipping..."
+            )
+            return
+
+        # Add script volume to cluster configuration
+        script_volume = V1Volume(
+            name=volume_name, config_map=V1ConfigMapVolumeSource(name=configmap_name)
+        )
+        self.volumes.append(script_volume)
+
+        # Add script volume mount to cluster configuration
+        script_mount = V1VolumeMount(name=volume_name, mount_path=mount_path)
+        self.volume_mounts.append(script_mount)
+
+        logger.info(
+            f"Added script volume '{configmap_name}' to cluster config: mount_path={mount_path}"
+        )
+
```

Comment:
This is a bit of a nit / a just-in-case but ConfigMaps have a 1MB limit for the data so might be a good idea to throw in a small check to validate the size. I don't think users would run into this often unless with huge scripts but it's one of those cases that we aren't 100% of the nature of the data science use case so no harm adding it

```suggestion
def _validate_configmap_size(self, scripts: Dict[str, str]) -> None:
    total_size = sum(len(content.encode('utf-8')) for content in scripts.values())
    if total_size > 1024 * 1024:  # 1MB limit
        raise ValueError(...) 
```